controller/src/main/java/org/jboss/as/controller/AttributesXMLDescription.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.controller;
+
+import java.util.Collection;
+import java.util.Map;
+
+import javax.xml.stream.XMLStreamException;
+
+import org.jboss.dmr.ModelNode;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+/**
+ * XML description of attributes in a {@link org.jboss.as.controller.registry.Resource}. Allows parsing and persisting
+ * Resource's attributes XML configuration.
+ *
+ * @author <a href=mailto:tadamski@redhat.com>Tomasz Adamski</a>
+ */
+public interface AttributesXMLDescription {
```

Comment:
I'm not enthusiastic about adding this interface at this time as a publicly available extension point. It becomes an API we may have to support for a decade, and I don't believe that over the next 6 months we're going to have the bandwidth to properly evaluate whether it's a good fit for such a long commitment.

Do we have a specific, known use case for an impl of this other than the Ignore and Nest impls you have here? If not we should consider using an abstract class with a package-protected constructor instead of an interface (or some other approach with similar semantics).